Больше деталей

ArthurSkorzeny opened the pull request:
---
:mortar_board: [Больше деталей](https://up.htmlacademy.ru/javascript/26/user/2100191/tasks/9)

:boom: https://htmlacademy-javascript.github.io/2100191-kekstagram-26/2/

ArthurSkorzeny committed:
Update main.js

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,66 @@
+//Создание массива комментариев
+const createCommentObject = () => {
+  return {
+    id: getRandomPositiveInteger('0','600'),
```
Кавычки не нужны. Ты сейчас передаёшь в функцию строки. Строки разбираются и превращаются в цифры. Это как минимум лишние операции.

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,66 @@
+const createCommentObject = () => {
+  return {
+    id: getRandomPositiveInteger('0','600'),
+    avatar: 'img/avatar-' + getRandomPositiveInteger('1','6') + '.svg',
```
1. Лишние кавычки у цифр
2. Используй обратные кавычки и синтаксис шаблонных строк.

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,66 @@
+  };
+}; 
+
+const commentObjects = Array.from({length: 1}, createCommentObject);
```
Это должно быть функцией `createComments`

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,66 @@
+  };
+}; 
+
+ const photoObjects = Array.from({length: PHOTO_OBJECT_COUNT}, createPhotoObject);
```
Это должно быть функцией `createPhotos`

o-bender reviewed (changes requested):
```
Array.from({length: PHOTO_OBJECT_COUNT}, (sourceArrayItem, sourceArrayItemIndex) => {
    //sourceArrayItem элемент исходного массива. Т.к. ты используешь хак с {length: PHOTO_OBJECT_COUNT}. То в нём всегда будет приходить undefined
   // sourceArrayItemIndex порядковый номер элемента - вот это можешь использовать как id
})
```

ArthurSkorzeny committed:
Update main.js

Много изменений я не сделал, я так и не понял, как преобразовать метод создающий массив, в функцию. Мне нужно положить его внутрь функции и вывести данные или как? Я пытался сделать по всякому, у меня ничего не получилось, в интернете искал , тоже не нашел , у меня вообще нет идей 0_0

ArthurSkorzeny committed:
Update main.js

Все сделал,комментарии прошлого комита - это последствие отсутствия сна, я извиняюсь

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,67 @@
+  };
+}; 
+
+const commentObjects = () =>{  
```
В именах функций должен быть глагол. Назови как-нибудь вроде createComments

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,67 @@
+  };
+}; 
+
+const photoObjects = () =>{
```
В именах функций должен быть глагол. Назови как-нибудь вроде createPhotos

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,67 @@
+const createCommentObject = () => {
+  return {
+    id: getRandomPositiveInteger(0, 600),
+    avatar: 'img/avatar-' + getRandomPositiveInteger(1, 6) + '.svg',
```
Используй обратные кавычки 
```
`img/avatar-${getRandomPositiveInteger(1, 6)}.svg`
```

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,67 @@
+const createPhotoObject = () => {
+  return {
+    id: getRandomPositiveInteger(1, 25),
+    url: getRandomPositiveInteger(1, 25),
```
Тут явно забыл что-то. URL не может быть просто случайной цифрой.

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,67 @@
+}; 
+
+const commentObjects = () =>{  
+  let comment = Array.from({length: 1}, createCommentObject);
```
Пусть генерится штук по 5 комментов. Это в следующих заданиях пригодится.

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,67 @@
+//Создание основного массива//
+const createPhotoObject = () => {
+  return {
+    id: getRandomPositiveInteger(1, 25),
```
Так они по-прежнему не будут уникальными. 
```
const createPhotoObject = (id) => {
    return {
         id,
         url: и т.д.
    };
}

const createPhotos = () => Array.from({length: PHOTO_OBJECT_COUNT}, (_, id) => createPhotoObject(id))
```

С id для комментариев тоже самое надо сделать.

ArthurSkorzeny commented:
Как я не пытаюсь, я не могу решить задачу с id и url , пытался как-то с вашей подсказкой работать, у меня ничего не получается и я не могу найти что либо в интернете. Возможно это очень просто, но я третий день не могу это сделать

ArthurSkorzeny commented:
Я пробовал выводить данные массива через цикл, но цикл начинается заново, при каждом запуске функции, есть ли какой то способ продолжать цикл вместе с основной функцией( 25 раз) ?

ArthurSkorzeny committed:
Update main.js

ArthurSkorzeny committed:
Update main.js

ArthurSkorzeny committed:
Update main.js

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,65 @@
+  id++
+  return {
+    id,
+    url: 'photos/'+id+'.jpg',
```
используй обратные кавычки. как в avatar.

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,65 @@
+  };
+}; 
+
+const createPhotos = () => Array.from({length: PHOTO_OBJECT_COUNT}, (_, id) => createPhotoObject(id))
```
вместо id++ давай лучше так:
```
const createPhotos = () => Array.from({length: PHOTO_OBJECT_COUNT}, (_, id) => createPhotoObject(id + 1))
```
И в комментариях тоже самое.

o-bender commented on js/main.js:
```diff
@@ -0,0 +1,65 @@
+  };
+};
+
+const createComment = () => Array.from({length: 5}, (_, id) => createCommentObject(id))
```
Лучше во множественном числе createComments

ArthurSkorzeny committed:
Update main.js

ArthurSkorzeny committed:
Merge branch 'master' into module4-task1
